pkg/controller/inferenceservice/resources/knative/service.go

```diff
@@ -111,6 +112,19 @@ func addLoggerAnnotations(logger *v1alpha2.Logger, annotations map[string]string
 	return false
 }
 
+func addLoggerContainerPort(container *v1.Container) {
+	if container != nil {
+		if container.Ports == nil {
```

Comment:
Why do container ports have to be nil? Can't we append if not?

Reply:
As per knative a servcie can only expose one user port. Refer to https://github.com/knative/serving/blob/24a4448a9d520e02aef1ebc3d041826d2b3819ff/pkg/reconciler/revision/resources/deploy.go#L174-L181, if multiple ports specified, only the first one will be picked up, so we cannot just append the port. In fact there's potential issue if another port rather than 8080 is used for predictor service, for logger now we just hard coded 8080(we have an option but not use it for now, may configure it to configmap in future), so if user used another port then the callservice function will not work. Think we may handle it when we do GRPC support in future, I can log an issue for it.